### ISSUE TITLE
Performance of XML beautify

### ISSUE THREAD

**Issue by codebude** [dbeck121/ConVista-CPI-Helper-Chrome-Extension#2]
The new beautifier feature is very nice. Unfortunately for bigger XML files the performance is really bad. My browser did hang for minutes.

I traced down the problematic code section which leads to the slow downs. It is the [encodeHtml](https://github.com/dbeck121/ConVista-CPI-Helper-Chrome-Extension/blob/0d36a0339e24842dd5359a4a3ab92244b16604a7/contentScript.js#L330)-function of the content-script:
```javascript
var buf = [];
  for (var i = str.length - 1; i >= 0; i--) {
    buf.unshift(['&#', str[i].charCodeAt(), ';'].join(''));
  }
return buf.join('')
```
I tried to re-write the function with a more performance-oriented coding. The outcome are two versions of the function.

One, that reproduces the exact same output, but with a better performance:
```javascript
function encodeHtmlMapJoin(str){
    return '&#'+str.split('').map(x=>x.charCodeAt()).join(';&#')+';'
}
```

And one with even higher performance, but a slightly different output. (But I think the output of this function still should work in the context of your plugin.)
```javascript
function encodeHtmlRegex(str){
    return String(str).replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;').replace(/"/g, '&quot;').replace(/\n/g,'&#010;').replace(/'/g, "&#039;")
}
```

I also did a quick benchmark on all three functions. You can [find the benchmark over here](https://es6console.com/k8g1oshl/). The results are as follows:
```
5000 calls to encodeHtmlBufferUnshift took 23163.539999979548 milliseconds. 
5000 calls to encodeHtmlMapJoin took 2499.1200000513345 milliseconds. 
5000 calls to encodeHtmlRegex took 614.8850000463426 milliseconds. 
```
So the re-written function with identical output should bring an improvement of round about ~10x. The slightly different one should bring an improvement by ~40x.

Feel free to take the above code for your plugin.

**Comment by dbeck121** [comment]
Hi @codebude 

That sounds good. The encodeHTML function was a quick and dirty solution. I tested the regex version from you and it makes a good impression. I released 1.0.3 with your regex function. It will be on the Chrome Store in a few minutes.

**Comment by codebude** [comment]
Thanks for the quick feedback.
